### ISSUE TITLE
Endpoint option not working with "Flag provided by not defined: -ovh-endpoint"

### ISSUE THREAD

**Issue by deyoungsterdela** [yadutaf/docker-machine-driver-ovh#33]
Using the latest **docker-machine-driver-ovh-v1.1.0-1-linux-amd64** _(with Docker Machine Version 0.8.2)_ and attempting to create a machine using a different endpoint _(i.e. Canadian OVH at `ovh-ca`)_ with the following command results in the below message `"flag provided but not defined: -ovh-endpoint"`;

```
root@c2922575bf9a:/# docker-machine -D create --ovh-region "GRA1" --ovh-flavor "s1-2" --ovh-image "Ubuntu 16.04" --ovh-ssh-user "ubuntu" --ovh-application-key "XXX" --ovh-application-secret "XXX" --ovh-consumer-key "XXX" --ovh-endpoint "ovh-ca" --ovh-project "Rancher" --ovh-ssh-key "ubuntu-rancher01" --driver ovh host-test10

Docker Machine Version:  0.8.2, build b98be6e
Found binary path at /usr/local/bin/docker-machine-driver-ovh
Launching plugin server for driver ovh
Plugin server listening at address 127.0.0.1:41443
() Calling .GetVersion
Using API Version  1
() Calling .SetConfigRaw
() Calling .GetMachineName
(flag-lookup) Calling .GetMachineName
(flag-lookup) Calling .DriverName
(flag-lookup) Calling .GetCreateFlags
Incorrect Usage.

Usage: docker-machine create [OPTIONS] [arg...]

Create a machine

Description:
   Run 'docker-machine create --driver name' to include the create flags for that driver in the help text.

Options:
   
   --driver, -d "none"											Driver to create machine with. [$MACHINE_DRIVER]
   --engine-env [--engine-env option --engine-env option]						Specify environment variables to set in the engine
   --engine-insecure-registry [--engine-insecure-registry option --engine-insecure-registry option]	Specify insecure registries to allow with the created engine
   --engine-install-url "https://get.docker.com"							Custom URL to use for engine installation [$MACHINE_DOCKER_INSTALL_URL]
   --engine-label [--engine-label option --engine-label option]						Specify labels for the created engine
   --engine-opt [--engine-opt option --engine-opt option]						Specify arbitrary flags to include with the created engine in the form flag=value
   --engine-registry-mirror [--engine-registry-mirror option --engine-registry-mirror option]		Specify registry mirrors to use [$ENGINE_REGISTRY_MIRROR]
   --engine-storage-driver 										Specify a storage driver to use with the engine
   --ovh-application-key 										OVH API application key. May be stored in ovh.conf [$OVH_APPLICATION_KEY]
   --ovh-application-secret 										OVH API application secret. May be stored in ovh.conf [$OVH_APPLICATION_SECRET]
   --ovh-billing-period "hourly"									OVH Cloud billing period (hourly or monthly). Default: hourly
   --ovh-consumer-key 											OVH API consumer key. May be stored in ovh.conf [$OVH_CONSUMER_KEY]
   --ovh-flavor "vps-ssd-1"										OVH Cloud flavor name or id. Default: vps-ssd-1
   --ovh-image "Ubuntu 16.04"										OVH Cloud Image name or id. Default: Ubuntu 16.04
   --ovh-project 											OVH Cloud project name or id
   --ovh-region "GRA1"											OVH Cloud region name
   --ovh-ssh-key 											OVH Cloud ssh key name or id to use. Default: generate a random name
   --ovh-ssh-user "ubuntu"										OVH Cloud ssh username to use. Default: machine
   --swarm												Configure Machine to join a Swarm cluster
   --swarm-addr 											addr to advertise for Swarm (default: detect and use the machine IP)
   --swarm-discovery 											Discovery service to use with Swarm
   --swarm-experimental											Enable Swarm experimental features
   --swarm-host "tcp://0.0.0.0:3376"									ip/socket to listen on for Swarm master
   --swarm-image "swarm:latest"										Specify Docker image to use for Swarm [$MACHINE_SWARM_IMAGE]
   --swarm-join-opt [--swarm-join-opt option --swarm-join-opt option]					Define arbitrary flags for Swarm join
   --swarm-master											Configure Machine to be a Swarm master
   --swarm-opt [--swarm-opt option --swarm-opt option]							Define arbitrary flags for Swarm master
   --swarm-strategy "spread"										Define a default scheduling strategy for Swarm
   --tls-san [--tls-san option --tls-san option]							Support extra SANs for TLS certs
flag provided but not defined: -ovh-endpoint
```

So there seems to be an issue where though the documentation indicates it as an option, the flag isn't actually factored in command, as the list of available options doesn't include `--ovh-endpoint`. Also, not sure if this is relevant, but the returned message says the undefined option is `-ovh-endpoint`, and not `--ovh-endpoint`, even though the option being passed is `--ovh-endpoint`.

If this is however by design, then it might be helpful to indicate that in the documentation, though then the command line approach for creating machines becomes quite limited then.

Cheers,
Dela.

**Comment by deyoungsterdela** [comment]
Just some additional info. Running the above command without the `--ovh-endpoint` flag rightly results in the 'invalid application key' error below, since the credentials are for the _OVH **Canada** service_;

```
Docker Machine Version:  0.8.2, build b98be6e
Found binary path at /usr/local/bin/docker-machine-driver-ovh
Launching plugin server for driver ovh
Plugin server listening at address 127.0.0.1:41229
() Calling .GetVersion
Using API Version  1
() Calling .SetConfigRaw
() Calling .GetMachineName
(flag-lookup) Calling .GetMachineName
(flag-lookup) Calling .DriverName
(flag-lookup) Calling .GetCreateFlags
Found binary path at /usr/local/bin/docker-machine-driver-ovh
Launching plugin server for driver ovh
Plugin server listening at address 127.0.0.1:36013
() Calling .GetVersion
Using API Version  1
() Calling .SetConfigRaw
() Calling .GetMachineName
(host-test12) Calling .GetMachineName
(host-test12) Calling .DriverName
(host-test12) Calling .GetCreateFlags
(host-test12) Calling .SetConfigFromFlags
Running pre-create checks...
(host-test12) Calling .PreCreateCheck
(host-test12) DBG | Validating billing period
(host-test12) DBG | Selecting billing period hourly
(host-test12) DBG | Validating project
Error with pre-create check: "Error 403: \"This application key is invalid\""
The default lines below are for a sh/bash shell, you can specify the shell you're using, with the --shell flag.

notifying bugsnag: [Error with pre-create check: "Error 403: \"This application key is invalid\""]
```

**Comment by yadutaf** [comment]
Hello @deyoungsterdela, Thanks for taking the time to report this issue! This flag is quite recent. It was added in commit c29f56d42aaa9ccfe7ca51e150372e0f81d77820. Can you confirm a clean install from master addresses your issue ? If so, I'll make sure to issue a release to close this issue.

**Comment by deyoungsterdela** [comment]
Sure thing @yadutaf . I'll give this a try and let you know how it goes. Thanks for your help. Cheers.

**Comment by deyoungsterdela** [comment]
Tested and verified that the new updated master works with the `--ovh-endpoint` flag _(and also fixes the related issue #34 using the `ovh.conf` file. I'll update and close that issue as well)_.

Thanks for the quick update and for your work on this @yadutaf. Really appreciated 👍
